neo/core/spiketrainlist.py

```diff
@@ -0,0 +1,294 @@
+        """Create a SpikeTrainList object from an array of spike times
+        and an array of channel ids."""
+        spike_time_array, dim = normalise_times_array(spike_time_array, units)
+        obj = cls()
+        obj._spike_time_array = spike_time_array
+        obj._channel_id_array = channel_id_array
+        obj._all_channel_ids = all_channel_ids
+        obj._spiketrain_metadata = {
+            "t_start": t_start,
+            "t_stop": t_stop
+        }
+        for name, ann_value in annotations.items():
+            if len(ann_value) != len(obj):
+                raise ValueError(f"incorrect length for annotation '{name}'")
+        obj._annotations = annotations
+        return obj
+
+    def _spiketrains_from_array(self):
+        """Convert multiplexed spike time data into a list of SpikeTrain objects"""
+        if self._spike_time_array is None:
+            self._items = []
+        else:
+            if self._all_channel_ids is None:
+                all_channel_ids = np.unique(self._channel_id_array)
+            else:
+                all_channel_ids = self._all_channel_ids
+            self._items = []
+            for i, channel_id in enumerate(all_channel_ids):
+                mask = self._channel_id_array == channel_id
+                times = self._spike_time_array[mask]
+                spiketrain = SpikeTrain(times, **self._spiketrain_metadata)
+                spiketrain.annotations = {
+                    name: value[i]
+                    for name, value in self._annotations.items()
+                }
+                # todo: consider adding channel id as metadata
+                spiketrain.segment = self.segment
+                self._items.append(spiketrain)
+
+    @property
+    def multiplexed(self):
+        """Return spike trains as a pair of arrays.
+
+        The first (plain NumPy) array contains the ids of the channels/neurons that produced
+        each spike, the second (Quantity) array contains the times of the spikes.
+        """
+        if self._spike_time_array is None:
+            # need to convert list of SpikeTrains into multiplexed spike times array
+            if self._items is None:
+                return np.array([]), np.array([])
+            else:
+                channel_ids = []
+                spike_times = []
+                dim = self._items[0].units.dimensionality
+                for i, spiketrain in enumerate(self._items):
+                    if hasattr(spiketrain, "load"):  # proxy object
+                        spiketrain = spiketrain.load()
+                    if spiketrain.times.dimensionality.items() == dim.items():
+                        # no need to rescale
+                        spike_times.append(spiketrain.times)
+                    else:
+                        spike_times.append(spiketrain.times.rescale(dim))
+                    channel_ids.append(i * np.ones(spiketrain.shape))
```

Comment:
we should force the dtype to int64 (the default is float64)

Reply:
done